src/com/google/enterprise/adaptor/filenet/FileNetAdaptor.java

```diff
@@ -83,46 +82,44 @@ public void initConfig(Config config) {
     config.addKey("filenet.username", null);
     config.addKey("filenet.password", null);
     config.addKey("filenet.objectStore", null);
+    config.addKey("filenet.objectFactory", null);
+    config.addKey("filenet.displayUrl", null);
+    config.addKey("filenet.isPublic", "false");
+    config.addKey("filenet.pushAcls", "true");
+    config.addKey("filenet.additionalWhereClause", "");
+    config.addKey("filenet.deleteAdditionalWhereClause", "");
+    config.addKey("filenet.excludedMetadata", "");
+    config.addKey("filenet.includedMetadata", "");
+    config.addKey("adaptor.namespace", Principal.DEFAULT_NAMESPACE);
   }
 
   @Override
   public void init(AdaptorContext context) throws Exception {
     this.context = context;
-    Config config = context.getConfig();
-
-    int maxFeedUrls = Integer.parseInt(config.getValue("feed.maxUrls"));
-    if (maxFeedUrls < 2) {
-      throw new InvalidConfigurationException(
-          "feed.maxUrls must be greater than 2");
-    }
-
-    contentEngineUrl = config.getValue("filenet.contentEngineUrl");
-    logger.log(Level.CONFIG, "filenet.contentEngineUrl: {0}", contentEngineUrl);
+    this.params = new Params(context.getConfig());
 
-    objectStore = config.getValue("filenet.objectStore");
-    logger.log(Level.CONFIG, "filenet.objectStore: {0}", objectStore);
-
-    username = config.getValue("filenet.username");
-    password = config.getValue("filenet.password");
-
-    // Verify we can connect to the server and access the ObjectStore.
-    logger.log(Level.INFO, "Connecting to content engine {0}",
-        contentEngineUrl);
     try (Connection connection = getConnection()) {
-      logger.log(Level.INFO, "Connecting to object store {0}", objectStore);
-      factory.getObjectStore(connection, objectStore);
+      logger.log(Level.INFO, "Connecting to object store {0}",
+          params.getObjectStore());
```

Comment:
I think the Params should get the factory class name, and hold both the actual ObjectFactory and possibly have getters for some of the created objects. FileSession in v3 had a getter for the ObjectFactory and created and passed the ObjectStore to the traversers. I think we want getConnection from this clas to move to Params as well. Some fuzziness abounds about the wiring, but getting the ObjectFactory and the Connection from the Params, and getting the rest either from the Params or the ObjectFactory (maybe depending on whether constructing them requires values from the Params?).